cdap-ui/app/ui-utils/url-generator.js

```diff
@@ -167,7 +167,9 @@ window.getOldCDAPUrl = function(navigationObj = {}) {
     'datasets.detail.overview.explore': '/datasets/:datasetId/overview/explore',
     'streams.detail.overview.status': '/streams/:streamId/overview/status',
     'streams.detail.overview.explore': '/streams/:streamId/overview/explore',
-    'apps.detail.overview.status': '/apps/:appId/overview/status'
+    'apps.detail.overview.status': '/apps/:appId/overview/status',
+    'apps.detail.overview.programs': '/apps/:appId/overview/programs',
+    'mapreduce.detail': '/apps/:appId/programs/mapreduce/:programId/runs'
```

Comment:
The programs are not just mapreduce, there are other programs (check for spelling whether they are plurals or singular):
- mapreduce
- worker
- service
- flow
- spark